### PR TITLE
Fix quality mode errors

### PR DESCRIPTION
**Pull request by ProfDoof**
# Summary
Fixes bug found [here](https://discord.com/channels/379378609942560770/379378610412191753/849006162786648094) with QualityMode being an unaccepted value by Discord. Also fixes a few problems with Naming Rule Violations.

# Details
There was a bug in the way that we were handling VideoQualityMode which caused channels to be created incorrectly when cloning channels. VideoQualityMode was not nullable. The couple of places that accepted a VideoQualityMode also used a default value of default instead of null which while it technically gives the desired behavior is not as clear as null is as to what the default value being put in is. 

# Changes proposed
* Switched from using `default` as the default value to using `null` for clarity. 
* Made `DiscordChannel#QualityMode` nullable so that when parsed it isn't set equal to 0 but instead null.
* Added testing commands to ensure that the changes didn't break anything. 
* Fix Naming Rule Violations

# Notes
N/A

### CODE REVIEW

File: DSharpPlus/Entities/Channel/Overwrite/DiscordOverwriteBuilder.cs [hunk 1]
```diff
@@ -129,14 +129,16 @@ public DiscordOverwriteBuilder For(DiscordRole role)
...
-
-            return this;
+            return (other.Type switch
+            {
+                OverwriteType.Member => new DiscordOverwriteBuilder(await other.GetMemberAsync().ConfigureAwait(false)),
+                OverwriteType.Role =>  new DiscordOverwriteBuilder(await other.GetRoleAsync().ConfigureAwait(false)),
+                _ => throw new NotImplementedException(),
+            })
+            .Allow(other.Allowed)
+            .Deny(other.Denied);
```
Comment (uwx) [inline DSharpPlus/Entities/Channel/Overwrite/DiscordOverwriteBuilder.cs:141]: This is a breaking change that doesn't add library features, it belongs in 5.x.

**Review by uwx — CHANGES_REQUESTED** [review]

File: DSharpPlus/Entities/Channel/Overwrite/DiscordOverwriteBuilder.cs (hunk 1 above)
Comment (ProfDoof) [reply DSharpPlus/Entities/Channel/Overwrite/DiscordOverwriteBuilder.cs]: I have no idea what you are talking about, this change doesn't exist. :joy:

**Review by uwx — APPROVED** [review]
LGTM

**Comment by abnormal0666** [comment]
This bug caused me a fricken headache let me tell ya.

**Review by Neuheit — APPROVED** [review]

**Pull request merged**